ceph/rados/monitor_workflows.py

```diff
@@ -281,3 +282,107 @@ def set_tiebreaker_mon(self, host) -> bool:
             return False
         log.info("Successfully set the new tiebreaker mon on cluster")
         return True
+
+    def connection_score_checks(
+        self, hosts, rados_obj, num_mons, mon_election_obj
+    ) -> bool:
+        """
+        Checks all mon daemons for correct number of monitors, correct number of peer monitors,
+        correct peer monitors exist for every monitor and valid connection scores for every
+        monitor ( connection scores in the range of 0 to 1 )
+
+        Example:
+            rados_obj = RadosOrchestrator(node=cephadm)
+            mon_election_obj = MonElectionStrategies(rados_obj=rados_obj)
+            mon_nodes = ceph_cluster.get_nodes(role="mon")
+            mon_nums = len(mon_nodes)
+
+            if not mon_workflow_obj.connection_score_checks(
+                mon_nodes, rados_obj, mon_nums, mon_election_obj
+            ):
+                raise Exception("Connection score checks failed")
+
+        Args:
+            hosts: Cluster object of all monitor host
+            rados_obj: RadosOrchestrator object
+            num_mons: Number of mons to exist in the cluster
+            mon_election_obj: MonElectionStrategies object
+
+        returns:
+            Pass -> True, Fail -> false
+        """
+        # Collect mon host and rank from ceph mon dump command
+        mon_quorum = mon_election_obj.get_mon_quorum()
+
+        for host in hosts:
+
+            if not rados_obj.check_daemon_exists_on_host(
+                host=host.hostname, daemon_type="mon"
+            ):
+                continue
```

Comment:
I feel that we can add a log statement, saying "Host X has no mon daemon present".

Reply:
Add log : "mon not present on host X, but mon label present on the host. Continuing onto the next host for testing" & the log level can be info or debug, since we are not failing the test case upon hitting the condition.